examples/config-many-files/teams.yml

```diff
@@ -0,0 +1,19 @@
+---
+apiVersion: v1
+kind: teams
+spec:
+  teams:
+  - name: workstations
+    hosts:
```

Comment:
Imagine I have a team (whether created by `fleetctl` or the UI).

In this team, I've created a secret (as shown in the `teams-enroll-secret.yml` example) that maps to the `workstations` team.

I assume that if I enroll with that secret, I'll get added to to the `workstations` team.

So... scenario:
* host_00001, host_00002, ..., host_0000n are currently enrolled, using the secret above

If I apply the example config file, will this remove all of the existing hosts from the team? Will they end up re-enrolling / being put back in the team by the enroll secret?

Manually specifying host names for the team seems more like an optional addition / manual override of hosts and adding them to a specific team. Is that a better way of thinking about it, rather than looking at the `hosts` here as an exhaustive list of hosts in a given team?

(I'm basing this on other concepts currently in fleet, such as packs, where if I have queryA, queryB in a pack, but then I apply a pack config with just queryA via fleetctl, it will stop scheduling queryB in the pack. If this is *different* behavior for different concepts, it will be important to make this clear in the documentation and possibly in the naming of this new concept.)

Reply:
>I assume that if I enroll with that secret, I'll get added to to the workstations team.

You're correct :)

> If I apply the example config file, will this remove all of the existing hosts from the team? Will they end up re-enrolling / being put back in the team by the enroll secret?

If you apply the example config file, the hosts that are specified as members of the `servers` team will be **transferred** from the `workstations` team to the `servers` team using a new API endpoint. The hosts that are specified as members so the `workstations` team will remain on the `workstations` team.

 > Is that a better way of thinking about it, rather than looking at the hosts here as an exhaustive list of hosts in a given team?

This is a great question. The current thinking is yes, specifying host names and users here acts as a manual override of hosts and users by transferring and adding them to a different team. As you've said this behavior is different from the current config concept in Fleet. I'm curious if it makes more sense to have this config file act as an exhaustive list. I think including all hosts and users on a team would be very difficult to maintain/update.

Reply:
> I think including all hosts and users on a team would be very difficult to maintain/update.

Very much agree, but wanted to confirm. I like the way you're thinking about the implementation of this one. It still would probably good to make this really clear in docs since it does differ from other fleet config concepts.